src/talemate/client/presets.py

```diff
@@ -89,150 +124,97 @@ def set_max_tokens(config: dict, kind: str, total_budget: int):
-        return PRESET_LLAMA_PRECISE
-    elif kind == "analyze_freeform_short":
-        return PRESET_LLAMA_PRECISE  # Assuming short freeform analysis uses the same preset as precise
-    elif kind == "narrate":
-        return PRESET_LLAMA_PRECISE
-    elif kind == "story":
-        return PRESET_DIVINE_INTELLECT
-    elif kind == "create":
-        return PRESET_TALEMATE_CREATOR
-    elif kind == "create_concise":
-        return PRESET_TALEMATE_CREATOR  # Assuming concise creation uses the same preset as creator
-    elif kind == "create_precise":
-        return PRESET_LLAMA_PRECISE
-    elif kind == "director":
-        return PRESET_SIMPLE_1
-    elif kind == "director_short":
-        return (
-            PRESET_SIMPLE_1  # Assuming short direction uses the same preset as simple
-        )
-    elif kind == "director_yesno":
-        return (
-            PRESET_SIMPLE_1  # Assuming yes/no direction uses the same preset as simple
-        )
-    elif kind == "edit_dialogue":
-        return PRESET_DIVINE_INTELLECT
-    elif kind == "edit_add_detail":
-        return PRESET_DIVINE_INTELLECT  # Assuming adding detail uses the same preset as divine intellect
-    elif kind == "edit_fix_exposition":
-        return PRESET_DETERMINISTIC  # Assuming fixing exposition uses the same preset as divine intellect
-    elif kind == "edit_fix_continuity":
-        return PRESET_DETERMINISTIC
-    elif kind == "visualize":
-        return PRESET_SIMPLE_1
+    config.update(preset_for_kind(kind, client))
+
+
+PRESET_MAPPING = {
+    "deterministic": PRESET_DETERMINISTIC,
+    "creative": PRESET_DIVINE_INTELLECT,
+    "simple": PRESET_SIMPLE_1,
+    "analytical": PRESET_ANALYTICAL,
+    "conversation": PRESET_TALEMATE_CONVERSATION,
+    "conversation_old": PRESET_TALEMATE_CONVERSATION,
+    "conversation_long": PRESET_TALEMATE_CONVERSATION,
+    "conversation_select_talking_actor": PRESET_TALEMATE_CONVERSATION,
+    "summarize": PRESET_LLAMA_PRECISE,
+    "analyze": PRESET_SIMPLE_1,
+    "analyze_creative": PRESET_DIVINE_INTELLECT,
+    "analyze_long": PRESET_SIMPLE_1,
+    "analyze_freeform": PRESET_LLAMA_PRECISE,
+    "analyze_freeform_short": PRESET_LLAMA_PRECISE,
+    "narrate": PRESET_LLAMA_PRECISE,
+    "story": PRESET_DIVINE_INTELLECT,
+    "create": PRESET_TALEMATE_CREATOR,
+    "create_concise": PRESET_TALEMATE_CREATOR,
+    "create_precise": PRESET_LLAMA_PRECISE,
+    "director": PRESET_SIMPLE_1,
+    "director_short": PRESET_SIMPLE_1,
+    "director_yesno": PRESET_SIMPLE_1,
+    "edit_dialogue": PRESET_DIVINE_INTELLECT,
+    "edit_add_detail": PRESET_DIVINE_INTELLECT,
+    "edit_fix_exposition": PRESET_DETERMINISTIC,
+    "edit_fix_continuity": PRESET_DETERMINISTIC,
+    "visualize": PRESET_SIMPLE_1,
+}
 
-    else:
-        return PRESET_SIMPLE_1  # Default preset if none of the kinds match
+
+def preset_for_kind(kind: str, client=None):
+    # Handle client-specific presets
+    if kind.startswith("conversation") and client == "tabbyapi":
```

Comment:
i want to keep direct references to specific clients out of  this file, so that clients can be self contained in their own repository if necessary. So instead of `client` being a client type string, it could just pass the client object itself and then check for and call a function on the client to the override.

however, i think this change becomes moot if i add some form of overridability to the presets - even if its just through manual editing of the config.yml file for now.

Reply:
This was primarily for my own testing the new sampler settings, I can revert it.